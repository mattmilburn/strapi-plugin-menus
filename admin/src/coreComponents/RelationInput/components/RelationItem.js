import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Box } from '@strapi/design-system/Box';
import { Flex } from '@strapi/design-system/Flex';

const ChildrenWrapper = styled(Flex)`
  width: 100%;
  /* Used to prevent endAction to be pushed out of container */
  min-width: 0;
`;

export const RelationItem = ({ children, disabled, endAction, style, ...props }) => {
  return (
    <Box style={style} as="li">
      <Flex
        paddingTop={2}
        paddingBottom={2}
        paddingLeft={4}
        paddingRight={4}
        hasRadius
        borderSize={1}
        background={disabled ? 'neutral150' : 'neutral0'}
        borderColor="neutral200"
        justifyContent="space-between"
        {...props}
      >
        <ChildrenWrapper justifyContent="space-between">{children}</ChildrenWrapper>
        {endAction && <Box paddingLeft={4}>{endAction}</Box>}
      </Flex>
    </Box>
  );
};

RelationItem.defaultProps = {
  disabled: false,
  endAction: undefined,
  style: undefined,
};

RelationItem.propTypes = {
  children: PropTypes.node.isRequired,
  disabled: PropTypes.bool,
  endAction: PropTypes.node,
  style: PropTypes.shape({
    height: PropTypes.number,
    left: PropTypes.number,
    position: PropTypes.string,
    right: PropTypes.number,
    width: PropTypes.string,
  }),
};
