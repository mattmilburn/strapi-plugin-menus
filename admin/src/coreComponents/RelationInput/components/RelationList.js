import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Box } from '@strapi/design-system/Box';
import { Stack } from '@strapi/design-system/Stack';

const ShadowBox = styled(Box)`
  position: relative;
  overflow-x: hidden;
  overflow-y: auto;

  &:before,
  &:after {
    position: absolute;
    width: 100%;
    height: 4px;
    z-index: 1;
  }

  &:before {
    /* TODO: as for DS Table component we would need this to be handled by the DS theme */
    content: ${({ overflowDirection }) =>
      overflowDirection === 'top-bottom' || overflowDirection === 'top' ? "''" : undefined};
    background: linear-gradient(rgba(33, 33, 52, 0.1) 0%, rgba(0, 0, 0, 0) 100%);
    top: 0;
  }

  &:after {
    /* TODO: as for DS Table component we would need this to be handled by the DS theme */
    content: ${({ overflowDirection }) =>
      overflowDirection === 'top-bottom' || overflowDirection === 'bottom' ? "''" : undefined};
    background: linear-gradient(rgba(0, 0, 0, 0) 0%, rgba(33, 33, 52, 0.1) 100%);
    bottom: 0;
  }
`;

export const RelationList = ({ children, overflow, ...props }) => {
  return (
    <ShadowBox overflowDirection={overflow} {...props}>
      <Stack spacing={1}>{children}</Stack>
    </ShadowBox>
  );
};

RelationList.defaultProps = {
  overflow: '',
};

RelationList.propTypes = {
  children: PropTypes.node.isRequired,
  overflow: PropTypes.oneOf(['top-bottom', 'bottom', 'top', '']),
};
