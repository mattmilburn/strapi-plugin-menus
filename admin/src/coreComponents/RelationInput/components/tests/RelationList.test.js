import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider, lightTheme } from '@strapi/design-system';

import { RelationItem } from '../RelationItem';
import { RelationList } from '../RelationList';

const setup = ({ endAction }) =>
  render(
    <ThemeProvider theme={lightTheme}>
      <RelationList>
        <RelationItem endAction={endAction}>First relation</RelationItem>
      </RelationList>
    </ThemeProvider>
  );

describe('Content-Manager || RelationInput || RelationList', () => {
  it('should render and match snapshot', () => {
    const { container } = setup({});

    expect(container).toMatchSnapshot();
  });
});
