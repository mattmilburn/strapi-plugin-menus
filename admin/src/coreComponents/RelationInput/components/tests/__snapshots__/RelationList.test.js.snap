// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content-Manager || RelationInput || RelationList should render and match snapshot 1`] = `
.c6 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c3 {
  background: #ffffff;
  padding-top: 8px;
  padding-right: 16px;
  padding-bottom: 8px;
  padding-left: 16px;
  border-radius: 4px;
  border-color: #dcdce4;
  border: 1px solid #dcdce4;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 {
  width: 100%;
  min-width: 0;
}

.c1 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 > * {
  margin-top: 0;
  margin-bottom: 0;
}

.c2 > * + * {
  margin-top: 4px;
}

.c0 {
  position: relative;
  overflow-x: hidden;
  overflow-y: auto;
}

.c0:before,
.c0:after {
  position: absolute;
  width: 100%;
  height: 4px;
  z-index: 1;
}

.c0:before {
  background: linear-gradient(rgba(33,33,52,0.1) 0%,rgba(0,0,0,0) 100%);
  top: 0;
}

.c0:after {
  background: linear-gradient(rgba(0,0,0,0) 0%,rgba(33,33,52,0.1) 100%);
  bottom: 0;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 c2"
      spacing="1"
    >
      <li
        class=""
      >
        <div
          class="c3 c4"
        >
          <div
            class="c4 c5"
          >
            First relation
          </div>
        </div>
      </li>
    </div>
  </div>
  <div
    class="c6"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
